tests/test_mixins.py

```diff
@@ -46,6 +111,32 @@ def test_create_should_return_different_instances_for_different_classes(
             AnotherSingletonDerived.create() is not SingletonDerived.create()
         )
 
+    def test_create_should_call_get_instance_key(self):
+        with patch.object(
+            SingletonClass, '_get_instance_key'
+        ) as get_instance_key_mock:
+            SingletonClass.create(1, 2, arg3=3)
+            get_instance_key_mock.assert_called_once_with(1, 2, arg3=3)
+
+    def test_create_should_pass_args_to_create_a_new_singleton_instance(self):
+        instance = SingletonClass.create(1, 2, arg3=3)
+
+        assert instance.arg1 == 1
+        assert instance.arg2 == 2
+        assert instance.arg3 == 3
+
+    def test_create_should_return_same_instance_for_equal_args(self):
+        instance_a = SingletonClass.create(1, 2, arg3=3)
+        instance_b = SingletonClass.create(1, 2, arg3=3)
+
+        assert instance_a == instance_b
+
+    def test_create_should_return_new_instance_for_different_args(self):
+        instance_a = SingletonClass.create(1, 2, arg3=3)
+        instance_b = SingletonClass.create(2, 1, arg3=3)
+
+        assert instance_a != instance_b
```

Comment:
It is better to use `is not`. The important thing here is not being the same instance.

```suggestion
        assert instance_a is not instance_b
```

Reply:
Fixed :+1: